SimCalorimetry/HGCalSimAlgos/interface/HGCalSiNoiseMap.h

```diff
@@ -62,36 +91,52 @@ class HGCalSiNoiseMap : public HGCalRadiationMap {
   std::vector<double> &getIleakParam() { return ileakParam_; }
   std::vector<std::vector<double> > &getENCsParam() { return encsParam_; }
   std::vector<double> &getLSBPerGain() { return lsbPerGain_; }
+  void setDefaultADCPulseShape(const std::array<float, 6> &adcPulse) { defaultADCPulse_ = adcPulse; }
+  const std::array<float, 6> &getADCPulseForGain(GainRange_t gain) {
+    if (ignoreGainDependentPulse_)
+      return defaultADCPulse_;
+    return adcPulses_[gain];
+  }
   std::vector<double> &getMaxADCPerGain() { return chargeAtFullScaleADCPerGain_; }
+  double getENCpad(const double &ileak);
+  void setCachedOp(bool flag) { activateCachedOp_ = flag; }
+
+  inline void setENCCommonNoiseSubScale(double val) { encCommonNoiseSub_ = val; }
 
 private:
+  GainRange_t defaultGain_;
+  int defaultAimMIPtoADC_;
+
+  //cache of SiCellOpCharacteristics
+  std::map<uint32_t, SiCellOpCharacteristicsCore> siopCache_;
```

Comment:
We're not using the siopCache_ at the moment because of the memory it would add (see our DPG talk): 50+ MB per core. If we were to use the caching, we'd want to keep its memory to the minimum  (hence ``std::map``) , while the computation which will be saved will vastly exceed the lower search performance with respect to ``std::unordered_map``

We keep the caching code available because in the next version of the geometry we'll be able to fold the detector and slash the memory by a factor 3 or even 6